cub/cub/device/dispatch/dispatch_select_if.cuh

```diff
@@ -420,12 +420,14 @@ template <typename InputIteratorT,
           typename EqualityOpT,
           typename OffsetT,
           bool KeepRejects,
-          bool MayAlias      = false,
-          typename PolicyHub = detail::select::policy_hub<cub::detail::value_t<InputIteratorT>,
-                                                          cub::detail::value_t<FlagsInputIteratorT>,
-                                                          detail::select::per_partition_offset_t,
-                                                          MayAlias,
-                                                          KeepRejects>>
+          bool MayAlias = false,
+          typename PolicyHub =
+            detail::select::policy_hub<detail::value_t<InputIteratorT>,
+                                       detail::value_t<FlagsInputIteratorT>,
+                                       OffsetT,
```

Comment:
important: this will degrade performance for H100 and earlier, if the offset type is 8 bytes wide. We should keep it `per_partition_offset_t`.

The reason we had it `detail::select::per_partition_offset_t` is that previously, tunings for the kernel were done for an `int` offset type. Then, we wanted to add support for larger number of items. So, we wanted to touch the kernel implementation as little as possible to not degrade performance. The solution was to use a streaming approach, that would keep the offset type of the kernel `int` and just add a little streaming context that provides information about previously selected items from previous partitions.
```suggestion
                                       detail::select::per_partition_offset_t,
```

Reply:
@gonidelis and I had a discussion on this recently and one of the problems may be that the tunings for the select and parition algorithms are going through the same policy hub here. But thx for providing that information, so we can make a better decision!

Reply:
Btw, we need to pass `OffsetT` here so we can tune independently for int32 and int64 on parition. If you look inside the policy hub, we have:

https://github.com/NVIDIA/cccl/pull/3617/files#diff-00b4790894d588f44f2f4511279f047870fce0afa24b5c6b094cda2c5f21e851R1589-R1591

```c++
    // We use KeepRejects to differentiate between partition and select in the tunings.
    // If KeepRejects is true we tune for partition otherwise we tune for select.
    static constexpr offset_size offset_t = KeepRejects ? classify_offset_size<OffsetT>() : offset_size::_4;
```

Reply:
Fixed the issue by parameterizing the policy hub with:
```
::cuda::std::conditional_t<SelectionOpt == SelectImpl::Partition, OffsetT, detail::select::per_partition_offset_t>
```